src/geometry/manifold/charted_space.lean

```diff
@@ -778,6 +779,34 @@ lemma singleton_has_groupoid (h : e.source = set.univ) (G : structure_groupoid H
 
 end singleton
 
+section open_subset
+variables (G : structure_groupoid H) [h : closed_under_restriction G] [has_groupoid M G]
+variables {s : set M} (hs : is_open s) [nonempty s]
```

Comment:
Instead of using an assumption that `s` is open, which is not accessible to typeclass inference, you could take `s` to be of type `topological_space.opens M`, i.e., a bundled open set (defined in `topology.opens`). Then the next def could be an instance.

Also, you shouldn't need the assumption that `s` is nonempty: if `s` is empty, then you can just take the empty atlas, and then there's no extension problem. A way to define the atlas could be, for instance,
```lean
atlas := ⋃ (x : M) (h : x ∈ s), 
    {@local_homeomorph.subtype_restr _ _ _ _ (chart_at H x) s.1 ⟨⟨x, h⟩⟩ s.2}
```